DataFormats/SiPixelDigiSoA/interface/SiPixelDigisDevice.h

```diff
@@ -15,11 +15,11 @@ class SiPixelDigisDevice : public PortableDeviceCollection<SiPixelDigisSoA, TDev
   SiPixelDigisDevice() = default;
   template <typename TQueue>
   explicit SiPixelDigisDevice(size_t maxFedWords, TQueue queue)
-      : PortableDeviceCollection<SiPixelDigisSoA, TDev>(maxFedWords + 1, queue) {}
+      : PortableDeviceCollection<SiPixelDigisSoA, TDev>(maxFedWords + 1, queue), nDigis_h(maxFedWords) {}
 
   // Constructor which specifies the SoA size
   explicit SiPixelDigisDevice(size_t maxFedWords, TDev const &device)
-      : PortableDeviceCollection<SiPixelDigisSoA, TDev>(maxFedWords + 1, device) {}
+      : PortableDeviceCollection<SiPixelDigisSoA, TDev>(maxFedWords + 1, device), nDigis_h(maxFedWords) {}
 
   void setNModulesDigis(uint32_t nModules, uint32_t nDigis) {
```

Comment:
Now that `nDigis` becomes part of the memory allocation in the `CopyToHost`, I'd suggest to remove it from here. Otherwise this function would open a very easy way to screw up the `memcpy()` (which I comment more in the next comment).

Reply:
Can we simply drop `nDigis_h` and let `nDigis()` return `metadata().size() - 1` ?

Reply:
Seems a solution to me (to drop `nDigis_h`), this would need some small fixes. I imagine also these updates in common between Run3 and Phase2 would benefit from a `14_0_X` back-port, no?

Reply:
> I imagine also these updates in common between Run3 and Phase2 would benefit from a `14_0_X` back-port, no?

Yes.